packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -2725,33 +2753,85 @@ export class ContainerRuntime
 			throw e;
 		}
 	}
+
+	/**
+	 * Processes messages that are not intended for the runtime layer.
+	 * The function today (July 2024) is identical to the 'processRuntimeMessage' function, but will progressively be updated to
+	 * to only having the functionality that is relevant to non-runtime messages.
+	 * @param messageWithContext - non-runtime messages to process with additional context and isRuntimeMessage prop as false
+	 */
+	private observeNonRuntimeMessage(
+		messageWithContext: MessageWithContext & { isRuntimeMessage: false },
+	) {
+		const { message, local } = messageWithContext;
+
+		// Intercept to reduce minimum sequence number to the delta manager's minimum sequence number.
+		// Sequence numbers are not guaranteed to follow any sort of order. Re-entrancy is one of those situations
+		if (
+			this.deltaManager.minimumSequenceNumber <
+			messageWithContext.message.minimumSequenceNumber
+		) {
+			messageWithContext.message.minimumSequenceNumber =
+				this.deltaManager.minimumSequenceNumber;
+		}
+
+		// Surround the actual processing of the operation with messages to the schedule manager indicating
+		// the beginning and end. This allows it to emit appropriate events and/or pause the processing of new
+		// messages once a batch has been fully processed.
+		this.scheduleManager.beforeOpProcessing(message);
+
+		this._processedClientSequenceNumber = message.clientSequenceNumber;
+
+		try {
+			// If there are no more pending messages after processing a local message,
+			// the document is no longer dirty.
+			if (!this.hasPendingMessages()) {
+				this.updateDocumentDirtyState(false);
+			}
+
+			this.emit("op", message, messageWithContext.isRuntimeMessage);
+
+			this.scheduleManager.afterOpProcessing(undefined, message);
+
+			if (local) {
+				// If we have processed a local op, this means that the container is
+				// making progress and we can reset the counter for how many times
+				// we have consecutively replayed the pending states
+				this.resetReconnectCount();
+			}
+		} catch (e) {
+			this.scheduleManager.afterOpProcessing(e, message);
+			throw e;
+		}
+	}
+
 	/**
 	 * Assuming the given message is also a TypedContainerRuntimeMessage,
 	 * checks its type and dispatches the message to the appropriate handler in the runtime.
 	 * Throws a DataProcessingError if the message looks like but doesn't conform to a known TypedContainerRuntimeMessage type.
 	 */
 	private validateAndProcessRuntimeMessage(
-		messageWithContext: MessageWithContext,
+		messageWithContext: MessageWithContext & { isRuntimeMessage: true },
 		localOpMetadata: unknown,
 	): void {
-		// TODO: destructure message and modernRuntimeMessage once using typescript 5.2.2+
-		const { local } = messageWithContext;
-		switch (messageWithContext.message.type) {
+		// TODO: destructure message and isModernRuntimeMessage once using typescript 5.2.2+
```

Comment:
typo: // TODO: destructure message and `isRuntimeMessage`

Reply:
Just delete the comment, you've done it :)